compiler/rustc_mir/src/interpret/intrinsics.rs

```diff
@@ -337,6 +338,19 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let result = Scalar::from_uint(truncated_bits, layout.size);
                 self.write_scalar(result, dest)?;
             }
+            sym::const_allocate => {
+                let size = self.read_scalar(args[0])?.to_machine_usize(self)?;
+                let align = self.read_scalar(args[1])?.to_machine_usize(self)?;
+
+                let align = match Align::from_bytes(align) {
+                    Ok(a) => a,
+                    Err(err) => bug!("align has to power of 2, {}", err),
```

Comment:
`bug!` isn't good here, this means users can cause ICEs. Please use `throw_ub_format!` instead.

Reply:
This should never be reached, right? We will be passing the values from AllocRef to here, right?

Reply:
You can add tests for all the error cases (e.g. allocating more than `isize::max_value()` or alignments that are not powers of two, ...). I think they can all be in one file